diff --git a/prefs.js b/prefs.js
index 4b8d986..2297c01 100644
--- a/prefs.js
+++ b/prefs.js
@@ -76,17 +76,7 @@ function init() {
 function get_schema(schema) {
     let extension = ExtensionUtils.getCurrentExtension();
 
-    // check if this extension was built with "make zip-file", and thus
-    // has the schema files in a subfolder
-    // otherwise assume that extension has been installed in the
-    // same prefix as gnome-shell (and therefore schemas are available
-    // in the standard folders)
-    let schemaDir = extension.dir.get_child('schemas');
-    let schemaSource;
-    if (schemaDir.query_exists(null))
-        schemaSource = GioSSS.new_from_directory(schemaDir.get_path(), GioSSS.get_default(), false);
-    else
-        schemaSource = GioSSS.get_default();
+    let schemaSource = GioSSS.new_from_directory("@gschemasDir@", GioSSS.get_default(), true);
 
     let schemaObj = schemaSource.lookup(schema, true);
     if (!schemaObj)
